### ISSUE TITLE
Outdated custom game configs

### ISSUE THREAD

**Issue by UE4SS** [UE4SS-RE/RE-UE4SS#348]
I noticed that the custom game configs haven't been updated alongside the default config, for example, the Jedi Survivor settings file.

Should we be updating these ? Maybe we've just completely forgotten about them ?

Should keep the signatures updated as well ? Because there's been no update to them for at least 5 months and it's possible games have updated and our custom files might not work.

If we should keep each config updated, how would we do that ?
Adding new items, and removing old ones from the ini file is easy enough but testing to see if each game still works is obviously more cumbersome and obviously not every dev might have access to every game that we have custom configs for.

**Comment by Buckminsterfullerene02** [comment]
Good point, these have indeed been forgetten about. Patternsleuth does still have support for the lua signature overrides but for the most part it's so good that it removes the need for many existing games to have them anymore. IIRC, in our database of roughly 1400 UE games, last report by Truman (3rd Jan) showed that there is >90% coverage for each AOB required by UE4SS.

![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/84156063/248cf8df-9ce9-4490-b335-90bf8419d5c2)

Regarding testing if they are required anymore - I can ask longer/truman/narknon (who have access to the database) to check which games we have sigs for work with patternsleuth and which don't - avoids the need of finding people with the game installed to check manually.

Once we know which work, we can remove the sigs. For the ones that don't work, we'll need someone to check if those custom sigs still work (unless ps has a built in utility for that, but I can't remember). And obviously do whatever needs to be done from there.

**Comment by Buckminsterfullerene02** [comment]
Tracking comment

**Checking if AOBs required**
- [x] Atomic Heart - need FNameToStringFString (probably inlined due to denuvo), StaticConstructObjectInternal (multiple addresses), EngineVersion
- [x] Borderlands 3 - no custom AOBs required
- [x] DeadIsland2 - no custom AOBs required
- [x] Final Fantasy 7 Remake - need GUObjectArray and FNameToStringFString
- [ ] Fuser - needs testing manually
- [x] Ghost Wire Tokyo - need GUObjectArray,  FTextFString  and EngineVersion 
- [x] Kingdom Hearts 3 - need GUObjectArray and FNameToStringFString and FNameCtorWchar 
- [x] Like a Dragon Ishin! - no custom AOBs required - see https://github.com/UE4SS-RE/RE-UE4SS/issues/348#issuecomment-3066727735
- [ ] Returnal - needs testing manually
- [ ] SCP 5K - needs testing manually
- [x] Satisfactory - see https://github.com/UE4SS-RE/RE-UE4SS/issues/348#issuecomment-1932551936
- [x] Star Wars Jedi Fallen Order - no custom AOBs required
- [x] Star Wars Jedi Survivor - no custom AOBs required
- [x] The Outer Worlds - no custom AOBs required
- [x] Walking Dead Saints & Sinners - need FNameToStringFString and StaticConstructObjectInternal
- [x] cyubeVR - no custom AOBs required

**Checking if existing AOBs still work for games that still need them**
- [ ] Atomic Heart - need FNameToStringFString (probably inlined due to denuvo), StaticConstructObjectInternal (multiple addresses), EngineVersion
- [ ] Final Fantasy 7 Remake - need GUObjectArray and FNameToStringFString
- [ ] Walking Dead Saints & Sinners - need FNameToStringFString and StaticConstructObjectInternal
- [ ] Ghost Wire Tokyo - need GUObjectArray,  FTextFString  and EngineVersion 
- [ ] Kingdom Hearts 3 - need GUObjectArray and FNameToStringFString and FNameCtorWchar

**Comment by Buckminsterfullerene02** [comment]
Will need to also remember to update the custom game configs docs on whatever we change

**Comment by UE4SS** [comment]
> Will need to also remember to update the custom game configs docs on whatever we change

It seems some of the links to the custom configs in the docs are broken right now.
Specifically the ones that don't directly link to a file but instead link to a location in the repo and it looks like those links weren't updated after the configs were moved into the assets directory.

**Comment by UE4SS** [comment]
Results for my manual test of Satisfactory Early Access, CL#273254 (latest as of now):
```
[17:35:17] PS Scan attempt 9671
[17:35:17] [PS] Reading image
[17:35:17] [PS] Starting scan
[17:35:17] [PS] Scan finished in 601.9╡s
[17:35:17] [PS] Failed to find EngineVersion: expected at least one value
[17:35:17] [PS] You need to override the engine version in 'UE4SS-settings.ini'.
[17:35:17] [PS] Failed to find GUObjectArray: expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/GUObjectArray.lua'
[17:35:17] [PS] Failed to find GMalloc: expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/GMalloc.lua'
[17:35:17] [PS] Failed to find FName::ToString: expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/FName_ToString.lua'
[17:35:17] [PS] Failed to find FName::FName(wchar_t*): expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/FName_Constructor.lua'
[17:35:17] [PS] Failed to find StaticConstructObject_Internal: expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/StaticConstructObject.lua'
[17:35:17] [PS] Failed to find FText::FText(FString&&): expected at least one value
[17:35:17] [PS] You can supply your own AOB in 'UE4SS_Signatures/FText_Constructor.lua'
[17:35:17] [PS] Scan failed

[17:35:17] Fatal Error: PS scan timed out
```

**Comment by Buckminsterfullerene02** [comment]
I kind of forgot, of course the existing configs won't work for SF, the game has been updated to UE5.2 lol

**Comment by UE4SS** [comment]
`Like a Dragon Ishin!`: No longer needed, all AOBs are found by the latest version of patternsleuth, so this config can be removed after we update.

**Comment by allaze-eroler** [comment]
look like someone forgot to include the GUOBkectArray.lua file for final fantasy 7 remake intergrade.... considering i got that problem:

<img width="979" height="512" alt="Image" src="https://github.com/user-attachments/assets/4f534355-6de2-4595-bf44-040343b19b7c" />

**Comment by UE4SS** [comment]
> look like someone forgot to include the GUOBkectArray.lua file for final fantasy 7 remake intergrade.... considering i got that problem:

Looks like you didn't check first: https://github.com/UE4SS-RE/RE-UE4SS/tree/main/assets/CustomGameConfigs/Final%20Fantasy%207%20Remake

Note that I can't vouch for this config still working, but it's clear from your screenshot that you haven't tried this config, so please try it, and if it doesn't work, make a new issue.

**Comment by allaze-eroler** [comment]
> > look like someone forgot to include the GUOBkectArray.lua file for final fantasy 7 remake intergrade.... considering i got that problem:
> 
> Looks like you didn't check first: https://github.com/UE4SS-RE/RE-UE4SS/tree/main/assets/CustomGameConfigs/Final%20Fantasy%207%20Remake
> 
> Note that I can't vouch for this config still working, but it's clear from your screenshot that you haven't tried this config, so please try it, and if it doesn't work, make a new issue.

I did before coming here, and it didn't work. Why do you think I shared that screenshot for? Besides, FF7remake got an update some months ago that changed a few things.

edit: ok, I think I know why, these files need to be put in the proper folder.... not in customgameconfigs but in the ue4ss_signatures folder.
